### ISSUE TITLE
Sprint: May 3 → May 15

### ISSUE THREAD

**Issue by RichardLitt** [ipfs/pm#105]
# Sprint May 3rd

Note that this is on **Tuesday**, not the usual Monday.

## Sprint Discussions

[Schedule](https://github.com/ipfs/pm#sprint-discussion-schedule)

Please take notes in a separate pad, if you can, and link it here.

Endeavour         | Lead            | Time (PDT - **UTC/Z** - CET) | Pad
:---------------: | :-------------: | :--------------------------: | :----:
sync              | @RichardLitt    | 9:00PDT **16:00Z** 18:00CEST  | IRC: #ipfs on Freenode
apps on ipfs      | @dignifiedquire | 10:30PDT **17:30Z** 19:30CEST | https://public.etherpad-mozilla.org/p/ipfs-may-3-apps-on-ipfs
infrastructure    | @lgierth        | 11:00PDT **18:00Z** 20:00CEST | https://public.etherpad-mozilla.org/p/ipfs-may-3-infrastructure
libp2p            | @diasdavid      | 11:30PDT **18:30Z** 20:30CEST | https://public.etherpad-mozilla.org/p/ipfs-may-3-libp2p
js-ipfs           | @diasdavid      | 12:00PDT **19:00Z** 21:00CEST | https://public.etherpad-mozilla.org/p/ipfs-may-3-js-ipfs
go-ipfs           | @whyrusleeping  | 12:30PDT **19:30Z** 21:30CEST | https://public.etherpad-mozilla.org/p/ipfs-may-3-go-ipfs

Please add the Agenda to the Pad before the endeavour sprint starts.

## Sprint Deliverables

- Add your tasks below in a comment, that way we only have people listed who are really in the sprint
- Add links to issues down here. Only add things you can finish this sprint.

**Comment by haadcode** [comment]
### @haadcode sprint goals - 3 May - 15 May
#### Summary
Make Orbit work as a stand-alone app (ie. without Node.js) in preparation for js-ipfs.

#### Tasks
##### orbit-db
  - [ ] Finish refactoring, add more tests, make a new release

##### Orbit
  - [ ] Make Orbit run in the browser as a stand-alone app (without Node.js)

##### CRDTs
  - [ ] Stretch goal: PN-Counter (increment/decrement counter)

##### PubSub
  - [ ] Test: https://github.com/ipfs/go-ipfs/pull/2599

**Comment by lgierth** [comment]
### @lgierth todo

- go-ipfs
  - [ ] CR https://github.com/jbenet/go-multiaddr-net/pull/15
  - [ ] CR https://github.com/ipfs/specs/pull/96
- infrastructure
  - [x] deploy SSL wildcard certificate
  - [ ] move domains to company account
  - [ ] look into multireq / v04x / IPNS issues ipfs/infrastructure#160
  - [ ] parallelize deploys
  - [ ] get `ssh <page>@pages.ipfs.team deploy <hash>` working
  - [ ] metrics: fix mtail's duplicate output
- DNS outage actionables
  - [ ] codify domains
  - [ ] prepare moving domains to gandi company account
  - [ ] better ipns tests
  - [ ] lower ttl for all ipfs.io zones
  - [ ] postmortem blog post
- berlin meetup
  - [x] do the meetup
  - [x] follow up with ligi about IPFSdroid
    - [x] URL scheme ligi/IPFSDroid#5
    - [x] xgo vs. gx karalabe/xgo#48

**Comment by noffle** [comment]
## noffle todo

- go-*
  - [x] review https://github.com/jbenet/go-multiaddr-net/pull/15
  - [x] review https://github.com/ipfs/go-libp2p/pull/38
- [x] ipfs-twitter-resolver (https://github.com/noffle/ipfs-twitter-resolver)
- js-libp2p-ipfs
 - [x] 100% coverage (https://github.com/ipfs/js-libp2p-ipfs/pull/7)

**Comment by RichardLitt** [comment]
#### todo   
- [ ] Work on Standard-Readme
- [ ] Write a blogpost about the Files API
- [ ] Publish Weekly 10
- [ ] Docs for https://github.com/ipfs/go-ipfs-api
- [ ] Rewrite programmtic default PR https://github.com/ipfs/go-ipfs/pull/2484/files

**Comment by dignifiedquire** [comment]
### todo

- [ ] ship js-ipfs-bitswap it an all its glory

**Comment by nginnever** [comment]
<h3>TODO</h3>
- <b>js-ipfs-api</b>
 - [ ] files get command
- <b>js-ipfs</b>
 - [ ] files get http-api
 - [ ] files get http-api tests
 - [ ] files get cli tests
 - [ ] files cat http-api
 - [x] files cat http-api tests
 - [x] files cat cli tests
 - [ ] files add http-api
 - [x] files add http-api tests
 - [x] files add cli tests
- hyperlog
 - [ ] replication algorithm write up
- distributed reddit clone
 - [ ] begin rework with js-ipfs
- pub/sub
 - [ ] test go-ipfs 1-> many tree
 - [ ] begin thinking about many -> many

**Comment by lgierth** [comment]
### @lgierth update

This sprint has been very slow and full of distractions for me.
Nevertheless, we had a great Berlin meetup at c-base, and finally have
SSL on all of ipfs.io.

pages.ipfs.team is on the home stretch, and same for the remaining fixes
to the infrastructure codebase, which will finally get it back to being
usable by everyone. These two (pages and infrastructure deploys) are
areas where I've been a bottleneck for the team.

That also means I'll finally have a free mind for taking on packet switching.

I striked through tasks which I realized where an over-commitment.

- go-ipfs
  - [ ] CR https://github.com/jbenet/go-multiaddr-net/pull/15
  - [ ] CR https://github.com/ipfs/specs/pull/96
- infrastructure
  - [x] deploy SSL wildcard certificate
  - [x] enable HSTS for SSL 
  - [x] move non-SSL sites to ipfs.team
    - mars.i.ipfs.io => mars.i.ipfs.team
    - metrics.i.ipfs.io => metrics.ipfs.team
    - ci.i.ipfs.io => ci.ipfs.team
  - [x] look into multireq / v04x / IPNS issues ipfs/infrastructure#160
  - [x] split v03x deploy into its own unit
  - [ ] parallelize deploys
  - [ ] in progress: get `ssh <page>@pages.ipfs.team deploy <hash>` working
  - [ ] ~~metrics: fix mtail's duplicate output~~
- DNS outage actionables
  - [x] prepare moving domains to gandi company account
  - [ ] move domains to company account
  - [ ] ~~codify domains~~
  - [ ] ~~better ipns tests~~
  - [ ] ~~lower ttl for all ipfs.io zones~~
  - [ ] ~~postmortem blog post~~
- berlin meetup
  - [x] do the meetup
  - [x] follow up with ligi about IPFSdroid
    - [x] URL scheme ligi/IPFSDroid#5
    - [x] xgo vs. gx karalabe/xgo#48

**Comment by RichardLitt** [comment]
### Update

Last week I broke down ipfs/go-ipfs#2484 into around thirty small PRs. This involved overhauling pretty much every option with a default possibility in the go-ipfs codebase. Most of these have been merged now, and I have also been actively trying to keep the http-api up-to-date in respect to this. Besides that, I also worked on the blog with @jbenet, published Weekly 10, organized some hangouts, and updated the READMEs for js-ipfs-api and go-ipfs-api. 

Finally, I did a lot of work this week on standard-readme: there is now a full [spec](https://github.com/RichardLitt/standard-readme) that **I want feedback on**, with what I think a good README should have. Extra IPFS-related rules can be found [here](https://github.com/RichardLitt/standard-readme/issues/9) (feedback also wanted). After talking with @noffle a lot, I also made a yeoman standard-readme [generator](https://github.com/RichardLitt/generator-standard-readme) that works well for scaffolding out your READMEs. Use it!

#### Done

- [x] Work on Standard-Readme
- [ ] Write a blogpost about the Files API
- [x] Publish Weekly 10
- [x] Rewrite programmatic default PR
- [~] Docs for https://github.com/ipfs/go-ipfs-api

**Comment by dignifiedquire** [comment]
@dignifiedquire update

The past week I have been exclusively working on js-ipfs. There were two main things that happened, the first one being the [merge of bitswap](https://github.com/ipfs/js-ipfs/pull/195) into js-ipfs and the second one being the [update of `ipfs object`](https://github.com/ipfs/js-ipfs/pull/219) to the newly formed [interface-ipfs-core](https://github.com/ipfs/interface-ipfs-core).

**Comment by diasdavid** [comment]
#### David's Update

I've been working solely focused on js-ipfs. There was a bunch of fixes, CR and debugging. The major/quantifiable things were:

- Create [`interface-ipfs-core`], write the Object API spec (plus the foundations to write the remaining api specs), create a batch of tests and upgrade js-ipfs-api (now 4.x.x release) with the new interface. This unblocks the possibility of swapping js-ipfs-api with js-ipfs for Orbit.
- Finish the multistream spec revision
- Update js-multistream to the latest revision of the multistream spec. I also started working on refreshing the API of the module to make it less wonky.

Other things involved the passing of the torch of ICWE to @dignifiedquire and giving a talk at require('lx'), the Node.js and JavaScript meetup.

**Comment by haadcode** [comment]
### @haadcode sprint end report - 3 May - 15 May
#### Summary
I planned to work on putting Orbit into the browser but ended up fixing and optimizing orbit-db for the most of the sprint. Good news is, the fixes & optimizations in orbit-db were absolutely worth it: orbit-db's write throughput performance is now at ~100ops/s (~33% improvement) and the code base has been fully modularized. Orbit now uses the following modules as datastores: orbit-db-eventstore, orbit-db-feedstore, orbit-db-kvstore, orbit-db-counterstore. I'm still in the process of updating all the documentation, though. Orbit also got some love and I've been integrating the new orbit-db version into it. Spent time also on code review & feedback for two new features to Orbit, implemented and PRed by two newcomers to the community: username auto-complete on 'tab' (by @masadeus) and emoji auto-complete and preview (by @shamb0t). Emoji auto-complete is not merged yet but will be in the next couple of days. I also spent some time on giving feedback to whyrusleeping's pubsub PR. Also tried to integrate js-ipfs to orbit-db (WIP).

#### Tasks
##### orbit-db
  - [X] Finish refactoring, add more tests, make a new release
  - [ ] Integrate with js-ipfs

##### Orbit
  - [ ] Make Orbit run in the browser as a stand-alone app (without Node.js)
  - [X] Username auto-complete: CR and merge
  - [ ] Emoji auto-complete/preview: CR and merge

##### CRDTs
  - [ ] Stretch goal: PN-Counter (increment/decrement counter)

##### PubSub
  - [X] Test: https://github.com/ipfs/go-ipfs/pull/2599
  - [X] Give feedback on pubsub ux

**Comment by Kubuxu** [comment]
### @Kubuxu sprint summary
Not much as I just started coding go-ipfs again, this time I hope that for longer and more actively. I've reviewed many PRs (hopefully making @whyrusleeping life at least a bit easier), started going through some outdated issues. My main focus was getting offline mode go-ipfs node up and running but it looks a tad harder that I thought it would be.

#### go-ipfs:
* [x] review many PRs (including ~15 of @RichardLitt)
* [x] start going through issues
* [x] PR: Copy short command description as long on: https://github.com/ipfs/go-ipfs/pull/2694
* [x] PR: Fix random t0060 failure: https://github.com/ipfs/go-ipfs/pull/2687
* [x] Start working on offline IPFS mode:
 * [x] Offline node starts
 * [x] Doesn't communicate to swarm
 * [x] Can fetch files via HTTP API
 * [ ] Adding files doesn't work as it supposed to
   * [x] Sharness test case in place
* [x] Start working on ipfs/go-ipfs#1394
 * [x] Sharness test case in place [ipfs/go-ipfs/feature/Fix-1394](https://github.com/ipfs/go-ipfs/tree/feature/Fix-1394)

**Comment by nginnever** [comment]
<h3>voxelot update</h3>

This week I did a bit of js-ipfs documentation work. Updated the unixfs-engine readme, started 'files' command documentation on interface-ipfs-core. Made dignifiedquire draw a flow chart :) 

I updated the exporter part of the engine a bit and now will be working to make the export a duplex object stream (progress bars!). Also working on tar archiving and gzip compression options. I continued to work on js-ipfs http and 'cat' is working, 'add' isn't returning the correct format to the go-http from the js http atm, and 'get' is still being implemented and is waiting on compression both in js-ipfs and js-ipfs-api.

**Comment by whyrusleeping** [comment]
## whyrusleeping update

This last week I spent a lot of time debugging and fixing the hanging issue with yamux. This is the issue that causes (in at least 90% of observed cases) the gateways and other nodes to hang when simple requests are made on them. I have a patch in place and am waiting on upstream code review to get it pushed in officially. 
During that process I also did a bunch of gx work, I refactored the dependency tree of go-stream-muxer to be more flat, making updating that area of the code easier, and also made `gx deps --tree` much more visually pleasing. 
I also fixed and pushed up a perf improvement on go-multiaddr. Previously to create a multiaddr from some bytes, we converted it into a string, and called `NewMultiaddr` on it which converted it back into bytes. This was eating up an unecessary amount of CPU time on the gateways.

This week I plan to continue on with go-libp2p fixes, and I also am planning on doing some light bitswap work. Next week I will be out on vacation so make sure you poke me soon if you need anything done before june.

**Comment by noffle** [comment]
## noffle update

Travel and being sick consumed a lot of my time this sprint. I still managed to review some go-libp2p PRs and make progress on some go-ipfs bugfixes. I also wrote an IPFS resolver that uses Twitter as a transient record store (ipfs-twitter-resolver). I also collaborated with @RichardLitt a bit on his standard-readme project and my own [common-readme](https://github.com/noffle/common-readme) project.

- go-ipfs, go-libp2p
  - [x] review https://github.com/jbenet/go-multiaddr-net/pull/15
  - [x] review https://github.com/ipfs/go-libp2p/pull/38
  - [~] files api cp/mv bugfix https://github.com/ipfs/go-ipfs/pull/2690
  - [~] flaky dht sharness test https://github.com/ipfs/go-ipfs/issues/2620
- alt resolvers
  - [x] ipfs-twitter-resolver (https://github.com/noffle/ipfs-twitter-resolver)
- js-libp2p-ipfs
 - [x] 100% coverage (https://github.com/ipfs/js-libp2p-ipfs/pull/7)

**Comment by jbenet** [comment]
### jbenet sprint update

The biggest direct things I handled this sprint were
- [x] grant proposals for IPFS
- [x] created project wide roadmap
- [x] community
  - [x] discussions with Akasha
  - [x] discussions with OB1/OpenBazaar
  - [x] discussions with Mediachain
  - [x] discussions with ConsenSys
  - [x] discussions with Blockstack
  - [x] gave a talk at Blockstack meetup
  - [x] Consensus Conf

Various upkeep things as usual:
- [x] lots of coordination with the various endeavours (go-ipfs, js-ipfs, etc)
- [x] some code review + merging
- [x] worked with richard on bumping/cleaning PRed blog posts

The bulk of my time is still consumed by organization management.
